cli/state.rs

```diff
@@ -94,6 +103,82 @@ impl ThreadSafeState {
   }
 }
 
+fn fetch_module_meta_data_and_maybe_compile_async(
+  state: &ThreadSafeState,
+  specifier: &str,
+  referrer: &str,
+) -> impl Future<Item = ModuleMetaData, Error = DenoError> {
+  let state_ = state.clone();
+  let specifier = specifier.to_string();
+  let referrer = referrer.to_string();
+
+  let f =
+    futures::future::result(ThreadSafeState::resolve(&specifier, &referrer));
```

Comment:
This is ok, but here's a tip: `Result` has an `IntoFuture` impl so you can do `ThreadSafeState::resolve(&specifier, &referrer).into_future().and_then(...)`.